docs/advanced-features/event-driven-packaging.md

```diff
@@ -0,0 +1,181 @@
+template during deployment:
+
+- `VPC`: Same as the existing Quilt instance.
+- `SecurityGroup`: Same as the existing Quilt instance.
+- `Subnets`: Same two subnets as the existing Quilt instance.
+- `BucketName`: Name of the Amazon S3 bucket to monitor. Must be
+read accessible to EDP.
+- `BucketIgnorePrefixes`: List of bucket path segments to ignore
+- `BucketPrefixDepth`: The number of `/`-separated path segments
+at the beginning of an S3 object key. Default value is `2`.
+- `BucketThresholdDuration`: Trigger a notification when this number
+of seconds has elapsed since the last object event in the S3 bucket 
+occurred. Default value is `300` seconds.
+- `BucketThresholdEventCount`: Trigger a notification when this
+number of files have been created (since the prior trigger)
+- `DBUser`: Username for EDP RDS instance
+- `DBPassword`: Password for EDP RDS instance
+- `EventBusName`: Name of custom EventBridge event bus that recieves events
+
+![](../imgs/edp-cloudformation-parameters.png)
+
+## How EDP works
+
+1. EDP monitors S3 object events for _s3://source-bucket_
+2. After a fixed number of events (`BucketThresholdEventCount`) or
+a maximum duration (`BucketThresholdDuration`) EDP creates a
+package in _s3://target-bucket_ (the _target_ bucket may or may not be
+the same as the _source_ bucket).
+3. Each EDP `files_ready` event contains sufficient information for the
+recipient to make Quilt data packages from the event:
+  - S3 Bucket name
+  - Common prefix
+  - Number of files
+  - Timestamp of event
+4. EDP ignores changes to ignored prefixes (`BucketIgnorePrefixes` parameter)
+in the source S3 bucket (e.g. `raw/*`)
+5. EDP publishes events to an AWS EventBridge bus and can forward
+events to external EventBridge buses
+6. Users subscribe the Amazon SNS topic created to recieve notifications
+
+## Example EDP events
+
+Currently, EDP generates two different `files_ready` events:
+- `package-data-ready`
+- `package-ready`
+
+### `package-data-ready`
+
+This event signals that a batch of files is ready to be packaged.
+For example with `p.set_dir(".",
+f"s3://{e['detail']['bucket']}/{e['detail']['prefix']}")`, followed
+by `p.push()`.
+
+<!--pytest.mark.skip-->
+```json
+{
+    "version":"0",
+    "id":"4a26374e-72f2-d78f-071b-dac895697670",
+    "detail-type":"package-objects-ready",
+    "source":"com.quiltdata.edp",
+    "account":"XXXXXXXXXXXX",
+    "time":"2022-12-08T20:01:34Z",
+    "region":"us-east-1",
+    "resources":[
+        "arn:aws:s3:::source-bucket"
+    ],
+    "detail":{
+        "version":"0.1",
+        "bucket":"source-bucket",
+        "prefix":"instrument-name/experiment-id/"
+    }
+}
+```
+
+### `package-ready`
+
+This event signals that a Quilt package is available in an Amazon
+S3 bucket and that a user-specified interval has elapsed to ensure
+that time-driven processes like File Gateway have synchronized their
+state to that of the S3 bucket.
```

Comment:
How exactly does this differ from the 'package created' event in Step 5 below?

Reply:
@drernie it's super janky but `package-ready` signals that a specified interval has passed so that e.g. StorageGateway RefreshCache must have been called. @robnewman please indicate the same and note that most users won't use this event.

Reply:
Added:
> This is used internally. Most users won't explicitly use this event.

Reply:
That's not correct either. This is not used internally. It's for users who want to know that a fixed time (we should note what that is) has passed and the files are now warm in e.g. Storage Gateway, as I explained above.